### DISCUSSION TITLE
GitHub does not render reStructuredText admonitions (notes, etc.) at all well

### DISCUSSION THREAD

**Discussion by rptb1** [community/community#45829] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

GitHub does a pretty good job of displaying reStructuredText documents, but it fails to render "[admonitions](https://docutils.sourceforge.io/docs/ref/rst/directives.html#admonitions)", such as notes.  For example, https://github.com/Ravenbrook/mps/blob/363eebf61b4d522f72143e38140afb65b6c6b3a2/design/collection.txt#L47-L54 
```
_`.framework.mpm`: The Collection Framework is merely a part of the
structure of the MPM. See design.mps.architecture and design.mps.arch
for the big picture.

.. note::

    Those two documents should be combined into one. Pekka P. Pirinen,
    1998-01-15.
```
is rendered as 
![image](https://user-images.githubusercontent.com/419807/215583646-af933fe9-3636-44e8-b9b2-90381ed245fd.png)
(and the HTML doesn't have the necessary markup either).

Admonitions don't need to be fancy, but they *do* need to stand out from the body text.

Thanks!

**Comment by flying-sheep** [comment]
See https://github.com/github/markup/issues/1682

**Reply by flying-sheep** [reply]
@zkoppert says that this discussion here is the right place to report this. Let’s hope he follows up.

**Comment by evdcush** [comment]
Related rST issues:

- https://github.com/orgs/community/discussions/86715
  - why rST will likely not get support: https://github.com/orgs/community/discussions/86715#discussioncomment-9706063

**Comment by rpdelaney** [comment]
From https://github.com/github/markup/issues/1682 :

> Thanks for the report, and thanks for the detailed examples.
> 
> After investigating, this repo's RST renderer (via docutils) **does** generate correct semantic HTML for admonitions — including proper `class="admonition note"` and `class="admonition warning"` attributes. The raw HTML output from this gem is correct.
> 
> However, the issue is in **step 2** of the rendering pipeline: GitHub's HTML sanitizer aggressively strips `class` and `id` attributes. Check out [this part of the README](https://github.com/github/markup#readme) for a fuller picture:
> 
> > The HTML is sanitized, aggressively removing things that could harm you and your kin — such as `script` tags, inline-styles, and `class` or `id` attributes.
> 
> Without those class attributes, the admonitions render as plain unstyled `<div>` elements. This is a downstream sanitization/styling issue that this repo can't address — the correct HTML is being generated, but it's stripped before display.
> 
> For admonition styling support, [GitHub's feedback discussions](https://github.com/orgs/community/discussions) would be the right venue to request that the sanitizer preserve or re-add admonition classes.
> 
> Closing as out of scope for this repository.

Vote the OP up, please!
